vcstool/commands/compare.py

```diff
@@ -1,80 +1,595 @@
 import argparse
-from cgitb import reset
-from distutils.util import execute
-from gettext import find
-from prettytable import PrettyTable
 import sys
-from unittest import result
-from vcstool import clients
-from vcstool.commands.branch import BranchCommand
-from vcstool.crawler import find_repositories
+import os
+import abc
+from typing import Any, Dict, List, Optional
+from enum import IntEnum, IntFlag, auto
+
+import prettytable as pt
 
-from vcstool.executor import ansi, execute_jobs, generate_jobs, output_repositories
+from vcstool.crawler import find_repositories
+from vcstool.executor import ansi, execute_jobs, generate_jobs
 from vcstool.commands.import_ import get_repositories
 from vcstool.streams import set_streams
+from vcstool.outputs import CompareOutput
 
 from .command import add_common_arguments
 from .command import Command
 
 
 class CompareCommand(Command):
 
-    command = 'compare'
-    help = 'Compare working copy to the repository list file'
+    command = "compare"
+    help = "Compare working copy to the repository list file"
 
-    def __init__(self, args):
-        super(CompareCommand, self).__init__(args)
+    def __init__(self, args: argparse.Namespace) -> None:
+        super().__init__(args)
+        self.progress = args.progress
+        self.workers = args.workers
+        self.debug = args.debug
 
+    @classmethod
+    def get_parser(cls) -> argparse.ArgumentParser:
+        parser = argparse.ArgumentParser(description=f"{cls.help}", prog="vcs compare")
+        group = parser.add_argument_group('"compare" command parameters')
+        group.add_argument(
```

Comment:
Nit: for consistency maybe just `--input` and `help=Where to read YAML from (default: -)` to follow `vcs import` functionality?

Reply:
Yep I can change that. A subtle difference here is we're reading the file as a string and not as an input buffer, which is how it's used elsewhere in the project.